Jenkinsfile

```diff
@@ -56,118 +95,105 @@ def stepClone()
     }
 }
 
-def stepPrintEnv(board, test)
+def stepRunNodeTests()
 {
-    sh 'dist/tools/ci/print_environment.sh'
+    catchError(buildResult: 'UNSTABLE', stageResult: 'FAILURE') {
+        def tests = []
+        stage( "${env.BOARD} setup"){
+            stepPrepareNodeWorkingDir()
+            tests = stepDiscoverTests()
+        }
+        for (int i=0; i < tests.size(); i++) {
+            stage("${tests[i]}") {
+                Exception caughtException = null
+                catchError(buildResult: 'UNSTABLE', stageResult: 'FAILURE') {
+                    try {
+                        stepPrintEnv()
+                        stepReset(tests[i])
+                        stepMake(tests[i])
+                        stepFlash(tests[i])
+                        stepTest(tests[i])
+                        stepArchiveTestResults(tests[i])
+                    } catch (org.jenkinsci.plugins.workflow.steps.FlowInterruptedException e) {
+                        // If we stop or something times out we want to exit the whole loop
+                        caughtException = e
+                    }
+                }
+                if (caughtException) {
+                    // This should exit out of the node that failed
+                    error caughtException.message
```

Comment:
Why don't we move this line into the `catch` statement? The surrounding `if` seems to be a bit verbose .. (below, too)

Reply:
Then it gets caught by the catcherror which sets build status and stage status. This is what is required to exit out.

I am willing to say there is a better way than using the catcherror call though.  I haven't tested.